Cli/EventListener/ApiErrorListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace SensioLabs\Insight\Cli\EventListener;
+
+use SensioLabs\Insight\Sdk\Exception\ApiClientException;
+use SensioLabs\Insight\Sdk\Exception\ApiParserException;
+use SensioLabs\Insight\Sdk\Exception\ApiServerException;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleErrorEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+
+class ApiErrorListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ConsoleEvents::ERROR => ['onConsoleError', 256],
+        ];
+    }
+
+    public function onConsoleError(ConsoleErrorEvent $event): void
+    {
+        $output = $event->getOutput();
+        $error = $event->getError();
+
+        if ($error instanceof ApiClientException) {
+            $this->showClientError($error, $output);
+        } elseif ($error instanceof ApiServerException) {
+            $output->writeln('<error>Server temporarily unavailable. Please try again in a few minutes.</error>');
+        } elseif ($error instanceof ApiParserException) {
+            $output->writeln('<error>Unable to process server response. Please try again later.</error>');
+        } elseif ($error instanceof TransportExceptionInterface) {
+            $output->writeln('<error>Network connection failed. Check your internet connection.</error>');
+        } else {
+            $output->writeln('<error>Something went wrong. Please try again.</error>');
+        }
+
+        $event->setExitCode(0);
```

Comment:
`0`?

Reply:
correct, but is that what ppl will expect? I'm not sure

Reply:
This doesn't make sense. Optimizing the displayed output shouldn't come at the price of breaking error codes. Can you figure out another way that doesn't break the exist code?